### ISSUE TITLE
Update request: kvmtool 2023-07-12 → New snapshot version

### ISSUE THREAD

**Issue by peigongdsd** [NixOS/nixpkgs#319611]
- Package name: kvmtool
- Latest released version:
2023-07-12-unstable
- Current version on the unstable channel: kvmtool does not have a release channel. Just follow master

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@astro @mfrw

Kvmtool-2023-07-12 does not run on Linux 6.9.3 #1-NixOS. It complains about some lacked features of KVM.
```
lkvm run -k /boot/EFI/nixos/kernel-6.9.3-43phjnq2svdisbrjsw27ilko7tccnfvo3ll7bfaio4vvguotngja.efi  -i /boot/EFI/nixos/initrd-6.9.3-5gy36mie4gpyzbgis2ljarxswu7lfb3hf6xd5tpxgaxqlsfx343a.efi --loglevel debug
  Info: # lkvm run -k /boot/EFI/nixos/kernel-6.9.3-43phjnq2svdisbrjsw27ilko7tccnfvo3ll7bfaio4vvguotngja.efi -m 1472 -c 20 --name guest-6001
KVM_CREATE_PIT2 ioctl: No such file or directory
```
However the master-branch version built directly from github does.
```
./lkvm run -k /boot/EFI/nixos/kernel-6.9.3-43phjnq2svdisbrjsw27ilko7tccnfvo3ll7bfaio4vvguotngja.efi  -i /boot/EFI/nixos/initrd-6.9.3-5gy36mie4gpyzbgis2ljarxswu7lfb3hf6xd5tpxgaxqlsfx343a.efi                 
  Info: # lkvm run -k /boot/EFI/nixos/kernel-6.9.3-43phjnq2svdisbrjsw27ilko7tccnfvo3ll7bfaio4vvguotngja.efi -m 1472 -c 20 --name guest-6098
PPrroobbiinngg  EEDDDD  ((eedddd==ooffff  ttoo  ddiissaabbllee))......  ookk

No EFI environment detected.
early console in extract_kernel
input_data: 0x0000000003ab12cc
input_len: 0x0000000000a9a8a5
output: 0x0000000001000000
output_len: 0x00000000034f6070
kernel_total_size: 0x0000000002a30000
needed_size: 0x0000000003600000
trampoline_32bit: 0x0000000000000000
Physical KASLR using RDRAND RDTSC...
Virtual KASLR using RDRAND RDTSC...

Decompressing Linux... No EFI environment detected.
Parsing ELF... Performing relocations... done.
Booting the kernel (entry_offset: 0x000000000000007a).
[    0.000000] Linux version 6.9.3 (nixbld@localhost) (gcc (GCC) 13.2.0, GNU ld (GNU Binutils) 2.41) #1-NixOS SMP PREEMPT_DYNAMIC Thu May 30 07:45:04 UTC 2024
[    0.000000] Command line: noapic noacpi pci=conf1 reboot=k panic=1 i8042.direct=1 i8042.dumbkbd=1 i8042.nopnp=1 earlyprintk=serial i8042.noaux=1 console=ttyS0 root=/dev/vda rw 
[    0.000000] BIOS-provided physical RAM map:
......
```

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by astro** [comment]
@peigongdsd Why not open a PR?

**Comment by peigongdsd** [comment]
> @peigongdsd Why not open a PR?

Not sure what version should I use, since there are other projects relying on kvmtool (like your microvm.nix) and also not sure whether the problem happens on others' machines or not. What do you think? Just take the snapshot of master branch?

**Comment by astro** [comment]
Exactly, I created the package with the latest available revision.

**Comment by astro** [comment]
Also, if you use and test kvmtool, consider adding yourself to the package's maintainers list.

**Comment by peigongdsd** [comment]
Okay, I'll do it.

**Comment by peigongdsd** [comment]
Hey bro, the url [kvm-tool-a-new-release](https://git.kernel.org/pub/scm/linux/kernel/git/will/kvmtool.git/snapshot/kvmtool-da4cfc3e540341b84c4bbad705b5a15865bc1f80.tar.gz) keeps returning 403 on my side. So does the older url you included in the package, namely [kvm-tool-original](https://git.kernel.org/pub/scm/linux/kernel/git/will/kvmtool.git/snapshot/kvmtool-106e2ea7756d980454d68631b87d5e25ba4e4881.tar.gz). Any idea on this?

**Comment by FliegendeWurst** [comment]
Fixed in #320221
